src/App.tsx

```diff
@@ -1,26 +1,114 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todosService from './api/todos';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { Filters } from './components/Filters/Filters';
+import { TodoError } from './components/TodoError/TodoError';
+import { ErrorMessages } from './types/Error';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentFilter, setCurrentFilter] = useState(Filters.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processedId, setProcessedId] = useState<number[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (!todosService.USER_ID) {
+      return;
+    }
+
+    setErrorMessage(null);
+    todosService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessages.UNABLE_TO_LOAD);
+        setTimeout(() => setErrorMessage(null), 3000);
+      });
+  }, []);
+
+  const handleCompletedStatus = (id: number) => {
+    const updatedTodos = todos.map(todo =>
+      todo.id === id ? { ...todo, completed: !todo.completed } : todo,
+    );
+
+    setTodos(updatedTodos);
+  };
+
+  const addTodo = (todo: Todo) => {
+    setTodos(prevTodos => [...prevTodos, todo]);
+  };
+
+  function deleteTodo(id: number) {
+    setProcessedId(ids => [...ids, id]);
+
+    return todosService
+      .deleteTodos(id)
+      .then(() => {
+        setTodos(currentTodos => currentTodos.filter(todo => todo.id !== id));
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.UNABLE_TO_DELETE);
+        setTimeout(() => setErrorMessage(null), 3000);
```

Comment:
You can create a handler to set and remove errors so you don't have to do it manually in each API call
```typescript
const handleError = (error: ErrorMessages) => {
  setErrorMessage(error);
  setTimeout(() => {
    setErrorMessage('');
  }, 3000);
};
```